### PR TITLE
dlFilter v1.17

### PR DESCRIPTION
**Pull request by Sophist-UK**
I have reverted the master branch and put my changes into a pull request instead, breaking it up into smaller commits that can more easily be code reviewed.

I have also added a ToDo list to the change log comment block to show what is still planned.

### CODE REVIEW

**Comment by Sophist-UK** [comment]
@SanderSade Feedback welcome. The enable / disable by group functionality may not have all the right event blocks in it, but aside from that this should be fully functional.

**Comment by SanderSade** [comment]
Going to look at the PR in a few, but I would go with v1.2 or even v2.0 for next release. Your fixes and updates merit a major update label, esp. when going with semver guidelines.

File: dlFilter.mrc [hunk 1]
```diff
@@ -566,91 +590,108 @@ on *:dialog:DLFilter_GUI:sclick:4: {
...
+    if (%DLF.custom.selected == Private notice) %DLF.custom.privnotice = $remtok(%DLF.custom.privnotice,%seltext,1,$asc($comma))
+    if (%DLF.custom.selected == Private ctcp) %DLF.custom.privctcp = $remtok(%DLF.custom.privctcp,%seltext,1,$asc($comma))
+    dec %selcnt
+  }
+  did -b DLF.Options.GUI 52
+  DLF.Options.SetCustomType $did(37).seltext
+  DLF.Options.SetRemoveButton
+}
+on *:dialog:DLF.Options.GUI:sclick:67: url -an http://dukelupus.com/dlfilter
+on *:dialog:DLF.Options.GUI:sclick:66: url -an http://www.dukelupus.pri.ee/download.php?f=187932020
```
Comment (SanderSade) [inline dlFilter.mrc]: Link both to GH, please

File: dlFilter.mrc [hunk 2]
```diff
@@ -1251,47 +1252,54 @@ alias CheckOpStatus {
...
+    if ((%br > 0) && ($gettok(%t,1,$asc($eq)) == DLFilter)) {
+      var %ver = $gettok(%t,2,$asc($eq))
+      if (%ver > $DLF.SetVersion) DLF.Update.StatusMsg Please update DLFilter to version %ver
+      elseif (%ver == $DLF.SetVersion) DLF.Update.StatusMsg You have current version of DLFilter
+      else DLF.Update.StatusMsg You have a newer version $br($DLF.SetVersion) than website $br(%ver)
+      .sockclose DLF.Update.Socket
+      return
     }
   }
+  DLF.Update.ErrorMsg Cannot find existing DLFilter version - please report this at http://www.dukelupus.com!
```
Comment (SanderSade) [inline dlFilter.mrc]: GH Issues page instead

**Comment by SanderSade** [comment]
I left some comments:
*  you should be listed everywhere as a co-author
* pretty much all website links should go to github
* separate changelog would be good

**Comment by Sophist-UK** [comment]
Review comments addressed (mostly).

Leaving as 1.17 for now - so we can have a v2 beta. But I agree in principle to changing to v2.

I prefer changelog in the script - at least whilst developing, and indeed I may make this visible in the About dialog when I write it.  Perhaps we can copy it to ReadMe.md when we publish it as a release?

**Comment by Sophist-UK** [comment]
P.S. Are you OK with the coding style?

**Comment by Sophist-UK** [comment]
I didn't think it was quite ready to merge, but if you want to change the version on your web site and point people to it, it can probably be released as beta.

**Comment by Sophist-UK** [comment]
P.S. In the future it might be better to merge as a sequence of commits rather than as a merged single commit.

If we ever need to track down the commit that has introduced a bug using git bisect, having separate commits would be useful.

**Comment by SanderSade** [comment]
Coding style seems to be OK - at least I didn't notice anything unseemly...

On commits - I prefer issue-based commits or PR's after initial major changes. That way there is a documented "trail" of changes and reasoning. But I am probably overthinking this for small things like dlf.

On beta - I would prefer not to have everybody switch to beta, but have some users ("testers") to try it first for a while. These are major changes, and if something rarely used/tested, like ops chat, breaks completely...

I probably can ask some people to try it out for a few days, but you have probably more options to do so, as I hardly ever use IRC these days.

**Comment by Sophist-UK** [comment]
My changes are not yet finished. So we can wait for a couple more weeks whilst I finish them.

Also, when you rewrite the ReadMe, can we add some guidelines on the information we need to receive if someone raises a support issue.

**Pull request merged**